packages/@react-stately/data/src/useTreeData.ts

```diff
@@ -159,7 +159,7 @@ export function useTreeData<T extends object>(options: TreeOptions<T>): TreeData
     if (newNode == null) {
       deleteNode(node);
     } else {
-      map.set(key, newNode);
+      addNode(newNode);
```

Comment:
So I followed the logic and this works, totally fine with merging this as is.
Thinking out loud, but all the extra addNode calls that result from a `move` operation when it calls 
```
// In move()
updateTree(items, toParentKey, parentNode...
``` 
kinda bother me since it will do a `map.set` for EVERY child of the parent node when all it needs to do is the following right?
1.  `map.set` for the updated parent node (to setup the new order of its child nodes)
2. `addNode` for the node that is being moved

A possible alternative would be to keep the above `map.set(key, newNode);` as is here and instead have 
```
 return updateTree(items, toParentKey, parentNode => {
          addNode(node);
          return ({
            key: parentNode.key,
            parentKey: parentNode.parentKey,
            value: parentNode.value,
            children: [
              ...parentNode.children.slice(0, index),
              node,
              ...parentNode.children.slice(index)
            ]
          })
        });
``` 
in `move()` perhaps? Or maybe we alter `deleteNode` to accept a secondary param (`deleteRecursive`) that turns off the recursive deletion so in the `move()` case we don't delete the node's children? All in all, prob ok to ignore if the recursive add isn't very expensive

Reply:
It's definitely a valid worry, a tree can get quite large. I was trying to think of how to minimize the deletion from the map as well. Good news is, this only operates on the subtree and this is the same complexity as it was before, which is why I decided this was an ok start.
Assuming it's got N children in the subtree, it's N calls to remove them (visit each node once and perform one operation on it), and then it's N calls to add them all back, which is only 2N which is equivalent to N as N gets large.
If things start suffering, we can try to do in place moves instead.

Reply:
hmm maybe I'm missing something, but wouldn't the child nodes of the node being moved already be in the map? why do they need to be added again?

Reply:
`move()` calls https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/data/src/useTreeData.ts#L312 which will fall into this `if` https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/data/src/useTreeData.ts#L159 and call delete, which will proceed to recursively delete all the children of the node https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/data/src/useTreeData.ts#L210